replica/tablets.cc

```diff
@@ -638,7 +646,8 @@ tablet_sstable_set::create_single_key_sstable_reader(
         mutation_reader::forwarding fwd_mr,
         const sstables::sstable_predicate& predicate) const {
     // The singular partition_range start bound must be engaged.
-    auto idx = group_of(pr.start()->value().token());
+    auto token = tablet_incremental_selector::find_starting_token(*this, pr.start()->value().token());
```

Comment:
I don't understand this.
If we find an sstable_set that doesn't contain the requested partition (it's a range but it should be singular) it will just return an empty reader, so why bother with `find_lowest_next_token`?
We can lookup the sstable_set and if it is not found, just return `make_empty_flat_reader_v2`.

Reply:
`find_sstable_set` hits `on_internal_error` if sstable set wasn't found. 
I think we could just return an empty sstable set instead

Reply:
Based on the discussion with @avikivity the internal_error is the desired behavior.